### ISSUE TITLE
[Redesign] MPAA Rating display

### ISSUE THREAD

**Issue by Nuklear92** [SerpentDrago/skin.auramod#164]
Okay, I have noticed that the MPAA rating on Movies as well as in TV Shows are a bit off when you're seeing them in addons like Seren, Venom, FEN, etc. I'll provide examples using Seren vs. TMDB Helper. However it also applies to to Venom and FEN which I also have installed.

As you'll see in the first screenshots this is how it looks on AuraMod. 

![Tmdb addon](https://user-images.githubusercontent.com/76958541/122954753-b6ef2300-d34d-11eb-9de4-cf333c2ca94f.jpeg)
![Seren addon](https://user-images.githubusercontent.com/76958541/122954863-cd957a00-d34d-11eb-9876-3b5d48891e0f.jpeg)

Now this is how it looks in a skin like Artic Horizon.

![TMDb addon](https://user-images.githubusercontent.com/76958541/122955116-ec940c00-d34d-11eb-8196-0915f1b5a9c5.jpeg)
![Seren addon](https://user-images.githubusercontent.com/76958541/122955262-f584dd80-d34d-11eb-87a1-907419bc70fc.jpeg)

I know they are both completely different skins and UIs. However, is there a possibility where this could get redesigned somehow it looks more uniform and doesn't look that off despite both addons displaying this in a different way. Example Arctic Horizon even though it displays the rating the same way it makes it look more uniform perhaps due to the separators after each info e.g;

**| 2021 | R | 1HR 59MINS |** and in the case of TMDb Helper **| 2021 | Rated R | 1HR 59MINS |**

In conclusion, I don't know if this is something that can be fixed or not due to the Netflix Style layout or better said Showcase layout. I just wanted to point this out since I haven't seen or noticed any other users doing any reference to it.

**Comment by SerpentDrago** [comment]
Step 5 in Post install instructions in the wiki covers this for TMDB helper , please read the install instructions. 
https://github.com/SerpentDrago/skin.auramod/wiki/Post-Install-Instructions

Will fix it for Tmdb helper . its not the skins problem if addons fill in that Meta info differently .  The convention should be to NEVER put "Rated" in the meta info list item . but some do . its up to that addon to have settings to stop this . 

Seren doesn't add a Prefix (like rated) to the MPAA rating , as it shouldn't . TMDB helper does by default but as stated this can be changed. any other addons should follow suite. 

it shouldn't be on the skins back to Detect and remove needless prefix's added that shouldn't be there to begin with .

**Comment by Nuklear92** [comment]
@SerpentDrago Probably I didn't made myself more clear, my apologies. I didn't meant the **Rated R** or **R**. I meant the design of it. If you notice in the screenshots the current design when a movie shows **R** it seems a bit off center and more to the left closed to eye icon. 

What I mean is if it's possible a small redesign of how the MPAA is displayed, maybe by eliminating the little box with the eye icon and just displaying the rating in plain letter similar to how Arctic Horizon does it. Or maybe rearrange the letter/s inside the little box more to the center so it looks nicer when showing **R** rating and I believe there are other MPAA ratings that also are displayed off center inside the box. If I'm not mistaken, **PG**, **NR**, **N/A** as not available are the others showing the same way. You can easily see this when checking out addons like Venom, FEN, Seren, etc. that doesn't add the Prefix as you pointed out.

**Comment by inb4after** [comment]
> @SerpentDrago Probably I didn't made myself more clear, my apologies. I didn't meant the **Rated R** or **R**. I meant the design of it. If you notice in the screenshots the current design when a movie shows **R** it seems a bit off center and more to the left closed to eye icon. 
> 
> What I mean is if it's possible a small redesign of how the MPAA is displayed, maybe by eliminating the little box with the eye icon and just displaying the rating in plain letter similar to how Arctic Horizon does it. Or maybe rearrange the letter/s inside the little box more to the center so it looks nicer when showing **R** rating and I believe there are other MPAA ratings that also are displayed off center inside the box. If I'm not mistaken, **PG**, **NR**, **N/A** as not available are the others showing the same way. You can easily see this when checking out addons like Venom, FEN, Seren, etc. that doesn't add the Prefix as you pointed out.

Agreed, this does need a visual improvement, especially the season and duration boxes etc.

I'll look into it after all the localization has been triple checked and all other bugs fixed.

**Comment by Nuklear92** [comment]
@inb4after @SerpentDrago thanks I appreciate it! 😃

**Comment by jojobrogess** [comment]
Having a little bit of trouble with the auto sizing, but I found it in the code:

https://user-images.githubusercontent.com/17692472/211319044-9d737837-e220-4705-acc4-f67ab3cf5f01.mp4

I'll post an update if I can get it to look right.

**Comment by jojobrogess** [comment]
Almost there:


https://user-images.githubusercontent.com/17692472/211445842-e173722c-2957-46f9-b859-9555bc85a92e.mp4



Just a spacing issue, then I have to test it with scrolling text.
 I did have to edit the image and redo the textures for it to work correctly lol

**Comment by jojobrogess** [comment]
Looks like I got it, some reason the video isnt showing the bottom of the flag box but its there.

https://user-images.githubusercontent.com/17692472/211474884-890cc03f-d3e8-484f-983a-1dea42138df2.mp4

All I have to do now is:
Slight bug check(check tv shows and aura to see if I broke anything),
technically redo the template because I did it wrong and rounded all the edges so it looks weird when you have it next to a box,
and then make a branch for it on my repo and PR it. 

I WILL SAY, that it looks kinda weird to see a box like that resize and leave all the space next to it...
Maybe I can clean that up. But then the area in which those icons sit would always fluctuate and be in somewhat different locations. UNLESS, I flipped the duration with the ratings. That way the end space doesn't matter.

**Comment by jojobrogess** [comment]
You know...
It just occured to me that this could have been mitigated by aligning the text to the center, rather than having the box resize based on the text length..

> ![Seren addon](https://user-images.githubusercontent.com/76958541/122954863-cd957a00-d34d-11eb-9876-3b5d48891e0f.jpeg)

I mean it still looks a little funky with all the space, cuz it's still there, it's just not inside the box.
![2023-01-11 02_36_40-Kodi](https://user-images.githubusercontent.com/17692472/211785005-0f11a791-bd2c-4042-97f2-20c94b4245a0.png)
idk I'm not liking my solution all that much.

Regardless, I ran into a problem with tv shows. It's overlapping like from my posts above. 
and
I still have to test out the scrolling feature, I have yet to find the setting where I can add characters to it so I can get it to scroll. 
So almost there.

**Comment by Nuklear92** [comment]
I commented long ago I believe it was to @inb4after about it on Discord or here in GH, don't remember quite well. And he said he had plans to make an actual redesign for this in particular, but I'm not sure if he still have the same plans for it or if AuraMod will actually go obsolete sooner or later and maintenance among UI improvements will completely stop eventually either for with Kodi Nexus or next major one v.21 Omega. 🤷‍♂️
